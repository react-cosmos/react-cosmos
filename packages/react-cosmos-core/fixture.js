module.exports = require('./dist/fixture/index.js');
