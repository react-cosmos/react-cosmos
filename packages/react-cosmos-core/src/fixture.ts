export * from './fixture/useSelect/index.js';
export * from './fixture/useValue/index.js';
export * from './fixture/Viewport.js';
