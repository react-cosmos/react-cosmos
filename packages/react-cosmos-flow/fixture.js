// @flow

import type { Node, ComponentType, ElementRef } from 'react';

export type FixtureType<P: {}> = {
  component: ComponentType<P>,
  name?: string,
  namespace?: string,
  props?: P,
  children?: Node, // Deprecated in favor of props.children
  init?: ({ compRef: ?ElementRef<ComponentType<P>> }) => Promise<any>
};

// Deprecated. Use this instead:
//  import { createFixture } from 'react-cosmos'
export function createFixture<P: {}, F: FixtureType<P>>(fixture: F) {
  return fixture;
}
