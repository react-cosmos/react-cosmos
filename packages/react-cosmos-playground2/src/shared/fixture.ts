export function removeFixtureNameExtension(fixtureName: string) {
  return fixtureName.replace(/\.(j|t)sx?$/, '');
}

export function removeFixtureNameSuffix(
  fixtureNameWithoutExtension: string,
  suffix: string
) {
  return fixtureNameWithoutExtension.replace(new RegExp(`\\.${suffix}$`), '');
}
