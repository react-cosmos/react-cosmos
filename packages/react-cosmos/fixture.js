require('regenerator-runtime/runtime');
module.exports = require('./dist/fixture');
