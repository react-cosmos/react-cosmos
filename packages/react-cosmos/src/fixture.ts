import 'regenerator-runtime/runtime';
export { Viewport, useValue } from 'react-cosmos-shared2/FixtureLoader';
