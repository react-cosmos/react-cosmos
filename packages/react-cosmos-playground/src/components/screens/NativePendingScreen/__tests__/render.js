import { create as renderer } from 'react-test-renderer';
import createContext from 'react-cosmos-test/generic';
import fixture from '../__fixtures__/default';

const { mount, getWrapper } = createContext({ renderer, fixture });

describe('NativePendingScreen', () => {
  beforeEach(mount);

  it('renders correctly', () => {
    expect(getWrapper().toJSON()).toMatchSnapshot();
  });
});
