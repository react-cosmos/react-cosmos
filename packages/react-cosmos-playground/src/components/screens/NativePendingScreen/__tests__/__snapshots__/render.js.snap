// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativePendingScreen renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="inner"
  >
    <img
      src={Object {}}
    />
    <div>
      <p
        className="header"
      >
        Standing by...
      </p>
      <p>
        <strong>
          Waiting for a signal from your React Native build.
        </strong>
      </p>
      <p>
        Make sure 
        <code>
          App.js
        </code>
         points to Cosmos and your app is running. See the
         
        <a
          href="https://github.com/react-cosmos/create-react-native-app-example"
          rel="noopener noreferrer"
          target="_blank"
        >
          CRNA example
        </a>
         
        for inspiration.
      </p>
      <p
        className="faded"
      >
        If something's wrong,
         
        <a
          href="https://github.com/react-cosmos/react-cosmos/issues/new"
          rel="noopener noreferrer"
          target="_blank"
        >
          report your error
        </a>
         
        and we'll do our best to help. Include as much details as you can.
      </p>
    </div>
  </div>
</div>
`;
